### ISSUE TITLE
Enhancement: require-await Add an option to ignore functions that return a `Promise`

### ISSUE THREAD

**Issue by oliveryasuna** [typescript-eslint/typescript-eslint#10368]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/require-await/

### Description

I propose that the `require-await` should have a boolean option that, when true, will not report unnecessary `async`s if the function explicitly returns a `Promise`.

For example, consider the following (simplified) snippet with `@typescript-eslint/require-await` set to `error`:

```typescript
async function foo(): Promise<void> {
  return;
}
```

ESLint reports "ESLint: Async function 'foo' has no 'await' expression.(@typescript-eslint/ require-await)".

Sometimes, it is good practice to mark your function as `async`, even if it does not use `await`, but still returns a promise. This, in my opinion, promotes writing intentional code.

I propose a new option, named something like `ignoreIfReturnPromise`, be added.

For example, with `'@typescript-eslint/require-await': ['error', {ignoreIfReturnPromise: true}]`:

```typescript
// This would NOT report.
async function foo(): Promise<void> {
  return;
}

// This would STILL report.
async function bar(): void {
  return;
}
```

### Fail

```TypeScript
Please see description above...
```

### Pass

```TypeScript
Please see description above...
```

### Additional Info

Thank you for considering my proposal.

I am happy to contribute, if someone could point me towards the best resources to get started on the knowledge necessary to make this addition.

**Comment by bradzacher** [comment]
The entire point of this rule is that it reports on the code you provided. It's intention is that it flags functions that might be unnecessarily async.

By creating a function that (a) does not have an await and (b) does not have a `return` whose expression is typed as a promise, you have violated the very premise of the rule.

For this reason - we shall reject this proposal.

---

> ```ts
> // This would STILL report.
> async function bar(): void { return; }
> ```

Note that this code is a semantic error in TS. All `async` functions must be typed to return a promise.
